src/features/textEditor/components/TipTapEditor.tsx

```diff
@@ -102,3 +108,26 @@ export function TipTapEditor({ editorContent, editorId, isActive, saveFileInMemo
     </div>
   );
 }
+
+export async function saveAsMarkdown(markdownSerializer: MarkdownSerializer, filePath: string) {
```

Comment:
I've moved this file here as he depends in the markdownSerializer that belong to the "text editor".
I've also simplified the code by using the new `getFileNameAndExtension`.

Reply:
After our chat I moved this to own file in `src/features/textEditor`